sdk/identity/identity/test/public/node/clientCertificateCredential.spec.ts

```diff
@@ -1,226 +1,118 @@
-      );
-    }
+    const token = await credential.getToken(scope);
+    assert.ok(token?.token);
+    assert.ok(token?.expiresOnTimestamp! > Date.now());
   });
 
-  it("sends a correctly formatted token request while tracing", async () => {
-    const tracer = new TestTracer();
-    setTracer(tracer);
-    const tenantId = "tenantId";
-    const clientId = "clientId";
-    const mockHttpClient = new MockAuthHttpClient();
-
-    const rootSpan = tracer.startSpan("root");
-
+  it("allows cancelling the authentication", async function() {
     const credential = new ClientCertificateCredential(
-      tenantId,
-      clientId,
-      path.resolve(__dirname, "../test/azure-identity-test.crt"),
-      mockHttpClient.tokenCredentialOptions
+      env.AZURE_TENANT_ID,
+      env.AZURE_CLIENT_ID,
+      certificatePath
     );
 
-    await credential.getToken("scope", {
-      tracingOptions: {
-        spanOptions: {
-          parent: rootSpan.context()
-        }
-      }
+    const controller = new AbortController();
+    const getTokenPromise = credential.getToken(scope, {
+      abortSignal: controller.signal
     });
 
-    rootSpan.end();
+    await delay(5);
+    controller.abort();
 
-    const rootSpans = tracer.getRootSpans();
-    assert.strictEqual(rootSpans.length, 1, "Should only have one root span.");
-    assert.strictEqual(rootSpan, rootSpans[0], "The root span should match what was passed in.");
+    let error: Error | undefined;
+    try {
+      await getTokenPromise;
+    } catch (e) {
+      error = e;
+    }
+    assert.ok(error instanceof CredentialUnavailable);
+    assert.ok(error?.message.includes("could not resolve endpoints"));
+  });
 
-    const expectedGraph: SpanGraph = {
-      roots: [
+  it("supports tracing", async function() {
+    if (isPlaybackMode()) {
+      // MSAL creates a client assertion based on the certificate that I haven't been able to mock.
+      // This assertion could be provided as parameters, but we don't have that in the public API yet,
+      // and I'm trying to avoid having to generate one ourselves.
+      this.skip();
+    }
+    await testTracing({
+      test: async (spanOptions) => {
+        const credential = new ClientCertificateCredential(
+          env.AZURE_TENANT_ID,
+          env.AZURE_CLIENT_ID,
+          certificatePath
+        );
+
+        await credential.getToken(scope, {
+          tracingOptions: {
+            spanOptions
+          }
+        });
+      },
+      children: [
         {
-          name: rootSpan.name,
-          children: [
-            {
-              name: "Azure.Identity.ClientCertificateCredential-getToken",
-              children: [
-                {
-                  children: [],
-                  name: "/tenantId/oauth2/v2.0/token"
-                }
-              ]
-            }
-          ]
+          name: "Azure.Identity.ClientCertificateCredential.getToken",
+          children: []
```

Comment:
I asked the other day about this to @richardpark-msft and I believe he said this was ok.

Relying on MSAL doesn't give me a way that I know of to pass through the abort signal all the way to the client, unless we create a new identity client (network layer for MSAL) per request.

We can do a lot more over time, but I'm hoping this is ok for starters (given that this is 2.0.0-beta.1)

Reply:
Oh I should mention, @richardpark-msft 's feedback in this case was that this tracing span, regardless of how layered are the internals, generally represents just one HTTP request (or less, if we can retrieve the token silently).